influxdb/src/main/java/com/hazelcast/jet/influxdb/InfluxDbSources.java

```diff
@@ -0,0 +1,343 @@
+     * @param database           name of the database
+     * @param connectionSupplier supplier which returns {@link InfluxDB} instance
+     * @param clazz              name of the POJO class
+     * @param <T>                type of the POJO class
+     * @param chunkSize          the number of {@link QueryResult}s to process in one chunk.
+     * @return <T> emits instances of POJO type
+     */
+    @Nonnull
+    public static <T> StreamSource<T> streamInfluxDb(@Nonnull String query, @Nonnull String database,
+                                                     @Nonnull SupplierEx<InfluxDB> connectionSupplier,
+                                                     @Nonnull Class<T> clazz, int chunkSize) {
+        checkTrue(query != null, "query cannot be null");
+        checkTrue(database != null, "database cannot be null");
+        checkTrue(connectionSupplier != null, "username cannot be null");
+        checkTrue(clazz != null, "clazz cannot be null");
+
+        return SourceBuilder.stream("influxdb-" + database,
+                ignored -> new InfluxDbStreamingSource<>(query, database, chunkSize, connectionSupplier, clazz))
+                .<T>fillBufferFn(InfluxDbStreamingSource::addToBufferWithMapping)
+                .destroyFn(InfluxDbStreamingSource::close)
+                .build();
+    }
+
+    /**
+     * A streaming source which executes a streaming query on InfluxDb and emits
+     * results as they arrive.
+     *
+     * @param <T> emitted item type
+     */
+    private static class InfluxDbStreamingSource<T> {
+
+        private final Class<T> clazz;
+        private final BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>(10000);
+        private final ArrayList<QueryResult> buffer = new ArrayList<>();
+        private final InfluxDBResultMapper resultMapper;
+        private InfluxDB db;
+        private volatile boolean finished;
+
+        InfluxDbStreamingSource(@Nonnull String query, @Nonnull String database, int chunkSize,
+                                @Nonnull SupplierEx<InfluxDB> connectionSupplier, @Nullable Class<T> clazz) {
+            this.clazz = clazz;
+            this.resultMapper = clazz != null ? new InfluxDBResultMapper() : null;
+            db = connectionSupplier.get();
+            db.query(new Query(query, database),
+                    chunkSize,
+                    queue::add,
+                    () -> finished = true
+            );
+        }
+
+        void addToBufferWithMapping(SourceBuffer<T> sourceBuffer) {
+            checkTrue(resultMapper != null, "resultMapper cannot be null!");
+            checkTrue(clazz != null, "clazz cannot be null!");
+            transferTo(result -> {
+                if (!result.hasError()) {
+                    resultMapper.toPOJO(result, clazz).forEach(sourceBuffer::add);
+                }
+            }, sourceBuffer);
+        }
+
+        void addToBufferWithOutMapping(SourceBuffer<Series> sourceBuffer) {
```

Comment:
I'm not sure if I like the idea of emitting `Series` objects, we should rather emit individual measurements in some format.